Dockerfile

```diff
@@ -15,16 +15,54 @@
 # along with Invenio; if not, write to the Free Software Foundation, Inc.,
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 
+###############################################################################
+## 1. Base (stable)                                                          ##
+###############################################################################
+
 FROM python:2.7
 
+# nodejs repo
+# detects distribution and adds the right repo
+# See: https://github.com/joyent/node/wiki/Installing-Node.js-via-package-manager#debian-and-ubuntu-based-linux-distributions
+RUN curl -sL https://deb.nodesource.com/setup | bash -
+
+# install requirements from repos
 RUN apt-get update && \
-    apt-get -qy install ssl-cert poppler-utils git subversion --fix-missing
+    apt-get -qy install poppler-utils git subversion nodejs mysql-client --fix-missing
+
+# install python requirements
+RUN pip install --upgrade pip && \
+    pip install mock
+
+# install nodejs requirements
+RUN npm update && \
+    npm install --silent -g bower less clean-css uglify-js requirejs
+
+
+###############################################################################
+## 2. Code (changing)                                                        ##
+###############################################################################
+
+# select proper requierments level
+#ENV REQUIREMENTS lowest
+#ENV REQUIREMENTS release
+ENV REQUIREMENTS devel
```

Comment:
For developers I would still use the latest released pypi package (which is likely newer than what is installed on production - see how packages are managed on Zenodo [here](http://zenodo.readthedocs.org/en/latest/development.html#python-requirements)).

The devel requirements are only used to detect possible problems with upcoming versions of dependent packages. These devel packages might have big fluctuations, and thus quite unreliable to dependent on for developers.